Improve bin packing concept page

lmktfy opened the pull request:
Improve the concept explanation of bin packing
- Use formulae per [style guide](https://kubernetes.io/docs/contribute/style/style-guide/#formulae-and-equations)
- Add page structure so that formulae make sense
- Don't make it so much about bin packing per se; explain that what Kubernetes provides is a quite rough approximation to ideal bin packing.
  _Counterargument_: humans packing actual bins also approximate an ideal solution,  rather than performing true NP-complete optimisation.
- Tweak a heading to align with the concept approach (it's not a task page)

:information_source: This was originally PR #49795; please also see the review history on that PR.

To make the idea work, this PR changes the styling for formulae on narrow viewports, so that they gain auto scroll bars.

[Original](https://kubernetes.io/docs/concepts/scheduling-eviction/resource-bin-packing/) vs [preview](https://deploy-preview-50664--kubernetes-io-main-staging.netlify.app/docs/concepts/scheduling-eviction/resource-bin-packing/)

lmktfy committed:
Add structure to resource bin packing concept page

lmktfy committed:
Scroll formulae on narrow viewports

lmktfy committed:
Use Docsy support for formulae

Co-authored-by: Bobby (Babak) Salamat <bsalamat@google.com>
Co-authored-by: Dipesh Rawat <rawat.dipesh@gmail.com>

lmktfy committed:
Tweak section heading

lmktfy commented:
/hold

Previous PR was held.

bsalamat reviewed (commented):
Thank you for the PR! Just a couple of minor points.

lmktfy commented on content/en/docs/concepts/scheduling-eviction/resource-bin-packing.md:
````diff
@@ -181,32 +180,60 @@ Used:
+```math
+  \begin{align}
+    { \text{intel.com/foo} } &\Longrightarrow { resourceScoringFunction((2 + 1), 4)} \tag*{} \\
+                             &= { 100 - ((4 - 3) \times { 100 \over 4 } )} \tag*{} \\
````
Revised

lmktfy committed:
De-emphasize "bin packing" in concept page

Kubernetes doesn't provide true bin packing out of the box; you would need
at least a custom scheduler and possibly also:

* a custom descheduler
* gang scheduling for the pending Pods
* readiness gates and other mechanisms

Ensuring ideal bin packing depends on being able to determine whether
the programs that might schedule will continue to run or halt, amongst
other properties. There are umpteen proofs that this is generally
undecidable. Anyway, we would know bin packing if we saw it, and this
is not it.

So, tweak the wording.

lmktfy commented:
I will also rebase against _main_.

lmktfy commented:
OK to unhold if the concern in https://github.com/kubernetes/website/pull/49795#issuecomment-2678370326 is addressed.

bsalamat commented:
/lgtm

Thank you!

divya-mohan0209 reviewed (commented):
More of a question as opposed to a suggestion - would sub-headings within examples (like [these](https://deploy-preview-50664--kubernetes-io-main-staging.netlify.app/docs/concepts/scheduling-eviction/resource-bin-packing/#node-scoring-n2-example-resource)) benefit from being in a bold font? While the current changes are in line with the style guidelines, the subheadings within the examples seem indistinct from the formulae.

lmktfy commented:
> the subheadings within the examples seem indistinct from the formulae.

I think they would look better done differently. Although the page overall would benefit from other improvements, I'd like to avoid an even bigger change — I find that when I propose a whole-page edit, reviews take a really long time.

lmktfy commented:
_Rebased_
